library/Mockery/Matcher/MatcherAbstract.php

```diff
@@ -25,7 +25,7 @@ abstract class MatcherAbstract
     /**
      * The expected value (or part thereof)
      *
-     * @var mixed
+     * @var array|mixed[]
```

Comment:
This change is wrong, and there are other places that need to also be corrected, like the constructor phpdoc.

Reply:
Hey @GrahamCampbell, sure I'll push a commit removing it. The purpose for the MR was really for the `andReturn` type hint.

However, could you please explain why? I'm wondering what I'm missing.

The phpdoc for the constructor is to provide type hints for consumers of that class, it has nothing to do with this property. At object instantiation, the constructor arguments are passed into an array, and only then we assign this new array to the local (protected) property.

https://www.php.net/manual/en/functions.arguments.php#functions.variable-arg-list.new

Reply:
This is indeed wrong. I mixed this constructor with another class. Nice catch. Fixed!